### ISSUE TITLE
How should electron build be handled?

### ISSUE THREAD

**Issue by chopfitzroy** [egoist/vuepack#115]
I noticed in the `package.json` there does not seem to be any reference to actually packaging up electron builds for distribution.

I am currently trying to wire something up using [electron-builder](https://github.com/electron-userland/electron-builder) but wondered if there was something I was missing?

Cheers.

**Comment by egoist** [comment]
You can use electron-builder, maybe we should add it to package.json later

**Comment by chopfitzroy** [comment]
Have been doing a bit of digging, might even submit a PR if I can get it going nicely but the current directory structure will require 2 `package.json`'s see this [issue](https://github.com/electron-userland/electron-builder/issues/572) for what I am talking about.

Do you think it would be worth a re-structure or just dropping in a temporary one for now?

**Comment by egoist** [comment]
Yep, I think 2-package.json structure is good, I use this structure everywhere too. We can put a `package.json` inside `./app`

**Comment by chopfitzroy** [comment]
Okay sweet as I will keep playing with this, and see if I can get something working nicely.

**Comment by adrianrudnik** [comment]
Pretty new to this stuff, is there any sample flow/package.json to for the app directory on how to compile a working app with electron-builder? **Edit**, got it working, though I'm not sure this would be the correct way:

```
app/package.json
{
  "name": "x",
  "version": "0.0.0",
  "author": "x",
  "description": "x",
  "devDependencies": {
    "cross-env": "^4.0.0",
    "electron-builder": "^17.0.3",
    "electron-prebuilt": "^1.4.13"
  },
  "build": {
    "appId": "xyz",
    "directories": {
      "output": "output"
    }
  },
  "scripts": {
    "clean": "rm -rf ./dist/linux-unpacked && rm -rf ./output",
    "build": "touch .env.prod && cross-env NODE_ENV=production build && rm .env.prod"
  },
  "main": "index.js",
  "files": [
    "dist/*.js",
    "dist/*.html",
    "dist/*.css"
  ]
}
```

Also a small change to the app/index.js so isDev is resolved differently for the elctron build:

```
const fs = require('fs')
if (fs.existsSync('.env.prod')) {
  process.env.NODE_ENV = 'production'
}

const isDev = process.env.NODE_ENV === 'development'
```

now a npm run build in the app directoy produces a valid and running AppImage in the app/output directoy.
